ui/src/app/views/cohort-edit/component.spec.ts

```diff
@@ -0,0 +1,267 @@
+
+import {Cohort, CohortListResponse} from 'generated';
+import {CohortsService} from 'generated';
+import { Observable } from 'rxjs/Observable';
+
+class Workspace {
+  id: string;
+  namespace: string;
+  cohorts: Cohort[];
+}
+
+function simulateInput(fixture: any, element: any, text: string) {
+  element.value = text;
+  element.dispatchEvent(new Event('input'));
+  updateAndTick(fixture);
+}
+
+function updateAndTick(fixture: any) {
+  fixture.detectChanges();
+  tick();
+}
+
+class CohortsServiceStub {
+  constructor() {
+    const stubWorkspace = new Workspace();
+    stubWorkspace.id = WorkspaceComponent.DEFAULT_WORKSPACE_ID;
+    stubWorkspace.namespace = WorkspaceComponent.DEFAULT_WORKSPACE_NS;
+
+    const exampleCohort: Cohort = {id: '',
+      name: '', description: '', criteria: '', type: ''};
+    exampleCohort.id = '1';
+    exampleCohort.name = 'sample name';
+    exampleCohort.description = 'sample description';
+
+    stubWorkspace.cohorts = [exampleCohort];
+    this.workspaces = [stubWorkspace];
+  }
+  public workspaces: Workspace[];
+
+  private getMatchingWorkspace(wsNamespace: string, wsId: string): Workspace {
+    return this.workspaces.find(function(workspace: Workspace) {
+      if (workspace.namespace === wsNamespace && workspace.id === wsId) {
+        return true;
+      } else {
+        return false;
+      }
+    });
+  }
+
+  public getCohort(wsNamespace: string,
+                   wsId: string,
+                   cId: string): Observable<Cohort> {
+    const observable = new Observable(observer => {
+      setTimeout(() => {
+        const workspaceMatch = this.getMatchingWorkspace(wsNamespace, wsId);
+        if (workspaceMatch !== null) {
+          observer.next(workspaceMatch.cohorts.find(function(cohort: Cohort) {
+            if (cohort.id === cId) {
+              return true;
+            }
+          }));
+          observer.complete();
+        }
+        observer.error(new Error('Error fetching a cohort. No cohort with \
+          id: ' + cId + ' exists in workspace ' + wsNamespace + ', ' + wsId +
+          ' in cohort service stub.'));
+      }, 0);
+    });
+    return observable;
+  }
+
+  public updateCohort(wsNamespace: string,
+                      wsId: string,
+                      cId: string,
+                      newCohort: Cohort): Observable<Cohort> {
+    const observable = new Observable(observer => {
+      setTimeout(() => {
+        const workspaceMatch = this.getMatchingWorkspace(wsNamespace, wsId);
+        if (workspaceMatch !== null) {
+          const index = workspaceMatch.cohorts.findIndex(function(cohort: Cohort) {
+            if (cohort.id === cId) {
+              return true;
+            }
+            return false;
+          });
+          if (index != null) {
+            workspaceMatch.cohorts[index] = newCohort;
+          }
+          observer.complete();
+        }
+        observer.error(new Error('Error updating a cohort. No cohort with id: '
```

Comment:
I think I agree that I'd prefer that. The one problem is that the observer will need to be passed into getMatchingWorkspace, which doesn't make a ton of sense from a function design standpoint to me? It may still be worth it though.